File: AsynchronousGrab/Forms/Source/FrameEventArgs.cs
/*=============================================================================
  Copyright (C) 2012 Allied Vision Technologies.  All Rights Reserved.
  Redistribution of this file, in original or modified form, without
  prior written consent of Allied Vision Technologies is prohibited.
-------------------------------------------------------------------------------
  File:        VimbaHelper.cs
  Description: Implementation file for the VimbaHelper class that demonstrates
               how to implement an asynchronous, continuous image acquisition
               with VimbaNET.
-------------------------------------------------------------------------------
  THIS SOFTWARE IS PROVIDED BY THE AUTHOR "AS IS" AND ANY EXPRESS OR IMPLIED
  WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF TITLE,
  NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A PARTICULAR  PURPOSE ARE
  DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
  AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
  TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
=============================================================================*/

namespace AsynchronousGrab
{
    using System;
    using System.Drawing;
    using AVT.VmbAPINET;
    /// <summary>
    /// EventArgs class that will contain a single image
    /// </summary>
    public class FrameEventArgs : EventArgs
    {
        /// <summary>
        /// The Image (data)
        /// </summary>
        private Image m_Image = null;
        private Frame m_Frame = null;
        /// <summary>
        /// The Exception (data)
        /// </summary>
        private Exception m_Exception = null;

        /// <summary>
        /// Initializes a new instance of the FrameEventArgs class. 
        /// </summary>
        /// <param name="image">The Image to transfer</param>
        public FrameEventArgs(Image image)
        {
            if (null == image)
            {
                throw new ArgumentNullException("image");
            }

            m_Image = image;
        }
        public FrameEventArgs(Frame frame)
        {
            if (null == frame)
            {
                throw new ArgumentNullException("frame");
            }

            m_Frame = frame;
        }
        /// <summary>
        /// Initializes a new instance of the FrameEventArgs class. 
        /// </summary>
        /// <param name="exception">The Exception to transfer</param>
        public FrameEventArgs(Exception exception)
        {
            if (null == exception)
            {
                throw new ArgumentNullException("exception");
            }

            m_Exception = exception;
        }

        /// <summary>
        /// Gets the image 
        /// </summary>
        public Image Image
        {
            get
            {
                return m_Image;
            }
        }
        public Frame Frame
        {
            get
            {
                return m_Frame;
            }
        }
        /// <summary>
        /// Gets the Exception
        /// </summary>
        public Exception Exception
        {
            get
            {
                return m_Exception;
            }
        }
    }
}

File: AVTGIGEascom/FrameEventArgs.vb
Imports System
Imports System.Drawing
Imports AVT.VmbAPINET

Public Class FrameEventArgs
        Inherits EventArgs

        Private m_Image As Image = Nothing
        Private m_Frame As Frame = Nothing
        Private m_Exception As Exception = Nothing

        Public Sub New(ByVal image As Image)
            If image Is Nothing Then
                Throw New ArgumentNullException("image")
            End If

            m_Image = image
        End Sub

        Public Sub New(ByVal frame As Frame)
            If frame Is Nothing Then
                Throw New ArgumentNullException("frame")
            End If

            m_Frame = frame
        End Sub

        Public Sub New(ByVal exception As Exception)
            If exception Is Nothing Then
                Throw New ArgumentNullException("exception")
            End If

            m_Exception = exception
        End Sub

        Public ReadOnly Property Image As Image
            Get
                Return m_Image
            End Get
        End Property

        Public ReadOnly Property Frame As Frame
            Get
                Return m_Frame
            End Get
        End Property

        Public ReadOnly Property Exception As Exception
            Get
                Return m_Exception
            End Get
        End Property
    End Class

